src/utilstrencodings.h

```diff
@@ -173,4 +174,24 @@ bool ConvertBits(const O& outfn, I it, I end) {
     return true;
 }
 
+/**
+ * Joins all the elements from the given input argument into a string object.
+ * @param[in] i             A vector with elements to join.
+ * @param[in] separator     A separator that joins the input elements.
+ * @return                  The joined elements.
+ */
+template<typename T>
+std::string Join(const std::vector<T>& i, const std::string& separator)
+{
+    std::ostringstream stream;
+    const auto end = i.cend();
+    for (auto it = i.cbegin(); it < end; ++it) {
+        stream << *it;
+        if (it < end - 1) {
+            stream << separator;
+        }
+    }
+    return stream.str();
+}
+
```

Comment:
Instead of a whole util function you might just inline a helper to append the warning:

```cpp
static void AppendWarn(std::string& res, const std::string& warn){
if (!res.empty()) res+=", ";
res+=warn;
}